lib/Raven/Breadcrumbs/Breadcrumb.php

```diff
@@ -0,0 +1,261 @@
+     */
+    const TYPE_ERROR = 'error';
+
+    /**
+     * @var string The category of the breadcrumb
+     */
+    private $category;
+
+    /**
+     * @var string The type of breadcrumb
+     */
+    private $type;
+
+    /**
+     * @var string The message of the breadcrumb
+     */
+    private $message;
+
+    /**
+     * @var string The level of the breadcrumb
+     */
+    private $level;
+
+    /**
+     * @var array The meta data of the breadcrumb
+     */
+    private $metaData;
+
+    /**
+     * @var float The timestamp of the breadcrumb
+     */
+    private $timestamp;
+
+    /**
+     * Constructor.
+     *
+     * @param string      $level    The error level of the breadcrumb
+     * @param string      $type     The type of the breadcrumb
+     * @param string      $category The category of the breadcrumb
+     * @param string|null $message  Optional text message
+     * @param array       $metaData Additional information about the breadcrumb
+     */
+    public function __construct($level, $type, $category, $message = null, array $metaData = [])
+    {
+        if (!in_array($level, self::getLevels(), true)) {
+            throw new InvalidArgumentException('The value of the $level argument must be one of the constants in Psr\Log\LogLevel.');
+        }
+
+        if (!in_array($type, self::getTypes(), true)) {
+            throw new InvalidArgumentException('The value of the $type argument must be one of the TYPE_* constants.');
+        }
+
+        $this->type = $type;
+        $this->level = $level;
+        $this->category = $category;
+        $this->message = $message;
+        $this->metaData = $metaData;
+        $this->timestamp = microtime(true);
+    }
+
+    /**
+     * Gets the breadcrumb type.
+     *
+     * @return string
+     */
+    public function getType()
+    {
+        return $this->type;
+    }
+
+    /**
+     * Gets the breadcrumb level.
+     *
+     * @return string
+     */
+    public function getLevel()
+    {
+        return $this->level;
+    }
+
+    /**
+     * Gets the breadcrumb category.
+     *
+     * @return string
+     */
+    public function getCategory()
+    {
+        return $this->category;
+    }
+
+    /**
+     * Gets the breadcrumb message.
+     *
+     * @return string
+     */
+    public function getMessage()
+    {
+        return $this->message;
+    }
+
+    /**
+     * Sets the breadcrumb message.
+     *
+     * @param string $message The message
+     *
+     * @return $this
+     */
+    public function setMessage($message)
+    {
+        $this->message = $message;
+
+        return $this;
+    }
+
+    /**
+     * Gets the breadcrumb meta data.
+     *
+     * @return array
+     */
+    public function getMetaData()
+    {
+        return $this->metaData;
+    }
+
+    /**
+     * Sets the breadcrumb meta data.
+     *
+     * @param array $metaData The meta data
+     *
+     * @return $this
+     */
+    public function setMetadata(array $metaData)
```

Comment:
Sometimes I find it written as `metaData`, sometimes as `metadata`, so I never understood which is the correct way, if there is one. Fixed, by the way